gnocchi/rest/api.py

```diff
@@ -579,12 +582,23 @@ def MetricSchema(cls, definition):
             else:
                 definition['archive_policy_name'] = ap.name
 
-        creator = pecan.request.auth_helper.get_current_user(
-            pecan.request)
+        resource_id = definition.get('resource_id')
+        if resource_id is None:
+            original_resource_id = None
+        else:
+            if name is None:
+                abort(400,
+                      {"cause": "Attribute value error",
+                       "detail": "name",
+                       "reason": "Name cannot be null "
+                       "if resource_id is not null"})
+            original_resource_id, resource_id = resource_id
 
         enforce("create metric", {
             "creator": creator,
             "archive_policy_name": archive_policy_name,
+            "resource_id": resource_id,
+            "original_resource_id": original_resource_id,
```

Comment:
we need this for policy check?

Reply:
Not with the actual policy rules, but if anyone wants to police after it, yes.